stm/stm.ml

```diff
@@ -1044,8 +1044,15 @@ end = struct (* {{{ *)
         match f acc (id, vcs, ids, tactic, undo) with
         | `Stop x -> x
         | `Cont acc -> next acc
- 
+
+  let undo_costly_in_batch_mode =
+    CWarnings.create ~name:"undo-batch-mode" ~category:"non-interactive" Pp.(fun v ->
+        str "Command " ++ Ppvernac.pr_vernac v ++
+        str (" is extremely costly in batch mode due" ^
+             " to a lack of a state cache for memory optimization reasons"))
+
   let undo_vernac_classifier v =
+    if !Flags.batch_mode then undo_costly_in_batch_mode v;
```

Comment:
I think this is wrong.  Should, e.g., `Abort All` really trigger this warning?  And `Reset Initial`?

Reply:
IMHO yes; I share @gares view about this warning really being about "meta" commands; in fact, it is likely that very soon we have an internal distinction between "meta" commands, such as `Undo`, `Abort` etc... that alter the global document vs `Vernaculars`, that behave locally.